modules/core/client/components/Board.js

```diff
@@ -0,0 +1,14 @@
+import { Children, cloneElement, useEffect } from 'react';
+import { selectPhoto } from '../services/photos.service';
+
+export default function Board({ name='bokeh', children, onNameChanged }) {
```

Comment:
I don't see anywhere where the name actually changes? I guess you implemented it for cases where it does change dynamically, although the current `tr-boards` directive seems to let you pass an array of names, and it will randomly select one and keep to that one, so no dynamic updating.

Reply:
> pass an array of names, and it will randomly select one

:+1: That's a correct observation, I seem to have ignored the _select random photo_ feature. I'll add a TODO comment for it. (We'll need it at Home page I think.)

> cases where it does change dynamically

Is it possible to implement a React component which doesn't change dynamically when its prop changes?

---

With the `useEffect` + `onNameChanged` some ancestor should tell another component to display credits of the photo.
- Now doing it angular way: Broadcasting/emitting some info to a $scope. (I don't understand details of what that means. Seems to work.)
- As the migration progresses, it will inform some shared state that this photo is displayed. It should also inform this state when the photo is not displayed anymore. I haven't implemented that.

So this component is somewhat confused. I'll have a closer look and fix it.

Does the above make sense?

Reply:
- allows array of photo names as input
- improve emitting info about displayed photos for credits (display/hide info; select photo when component mounts only)
- added the TODO for tr-boards-ignore-small

Implemented in 76d84e9c3e396256b181a84f3b55836b3edcac1a

Reply:
> Is it possible to implement a React component which doesn't change dynamically when its prop changes?

No, I doubt it, it's just that if it will never change, then the emitting the name change would never happen (apart from the initial call), but it's a bit odd that you tell it the name, then it emits back to you what the name is! ... however ....

> As the migration progresses, it will inform some shared state that this photo is displayed.

... because of this, I don't think we need to address that right now.

So, yes! Resolves issue! :star2: